daml-foundations/daml-ghc/daml-compiler/src/DA/Service/Daml/Compiler/Impl/Handle.hs

```diff
@@ -149,42 +146,39 @@ data DalfDependency = DalfDependency
 newtype GlobalPkgMap = GlobalPkgMap (Map.Map UnitId (LF.PackageId, LF.Package, BS.ByteString, FilePath))
 instance Shake.IsIdeGlobal GlobalPkgMap
 
-newIdeState :: Options
-            -> Maybe (Event -> IO ())
-            -> Logger.Handle IO
-            -- This type is rather awkward. The reason for this is that
-            -- we can only fully initialize the IdeState once we get the
-            -- LspFuncs in the initialize callback.
-            -- We should move away from Managed (we only use it for the scenario service)
-            -- and then refactor this.
-            -> Managed (VFSHandle -> IO IdeState)
-newIdeState compilerOpts mbEventHandler loggerH = do
-    mbScenarioService <-
-        for (guard (optScenarioService compilerOpts) >> mbEventHandler) $ \eventHandler ->
-            Scenario.startScenarioService eventHandler loggerH
-
+getIdeState
+    :: Options
+    -> Maybe Scenario.Handle
+    -> Logger.Handle IO
+    -> Maybe (Event -> IO ())
```

Comment:
Can we now simplify this to always being present, and people can pass `\x -> return ()` if they don't want to do something. I hated the fact we switched on it, so nice work fixing that bit.

Reply:
Good idea, changed!